telegram-desktop (and forks): make webkitgtk optional, and update to latest version

NickCao opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

NickCao committed:
telegram-desktop: make webkitgtk optional

ilya-fedin commented on pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/unwrapped.nix:
```diff
@@ -0,0 +1,150 @@
+      libicns
+    ];
+
+  dontWrapQtApps = true;
```
This does nothing as wrapQtAppsHook is not in nativeBuildInputs

NickCao replied:
If it's not set qtbase's setup hook would complain.

ilya-fedin replied:
ah, ok

ilya-fedin commented:
Can you also remove kotato's with-webkit.nix and the corresponding entry from all-packages.nix?

NickCao commented:
> Can you also remove kotato's with-webkit.nix and the corresponding entry from all-packages.nix?

Done. materialgram is still broken, maybe also switch it to qt5?

NickCao committed:
materialgram: 5.6.1.1 -> 5.7.0.1

Diff: https://github.com/kukuruzka165/materialgram/compare/refs/tags/v5.6.1.1...v5.7.0.1

Changelog: https://github.com/kukuruzka165/materialgram/releases/tag/v5.7.0.1

ilya-fedin commented:
> Done. materialgram is still broken, maybe also switch it to qt5?

Maybe... Another way is to switch tdesktop to ffmpeg_6 which should make tdesktop, materialgram and 64gram based on 5.7x build.

NickCao committed:
telegram-desktop: build with ffmpeg_6

NickCao committed:
telegram-desktop: 5.6.3 -> 5.7.1

Diff: https://github.com/telegramdesktop/tdesktop/compare/v5.6.3...v5.7.1

Changelog: https://github.com/telegramdesktop/tdesktop/releases/tag/v5.7.1

NickCao committed:
_64gram: 1.1.43 -> 1.1.45

Diff: https://github.com/TDesktop-x64/tdesktop/compare/v1.1.43...v1.1.45

Changelog: https://github.com/TDesktop-x64/tdesktop/releases/tag/v1.1.45

NickCao committed:
telegram-desktop.tg_owt: build with ffmpeg_6

NickCao commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353524`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>_64gram</li>
    <li>kotatogram-desktop</li>
    <li>materialgram</li>
    <li>telegram-desktop</li>
  </ul>
</details>

ilya-fedin commented:
hmm :thinking: 
```
nix-build --expr 'with import ./. { }; kotatogram-desktop.override { withWebkit = false; }'
error:
       … while calling a functor (an attribute set with a '__functor' attribute)

         at «string»:1:22:

            1| with import ./. { }; kotatogram-desktop.override { withWebkit = false; }
             |                      ^

       … while calling a functor (an attribute set with a '__functor' attribute)

         at /home/ilya/nixpkgs/lib/customisation.nix:144:43:

          143|       # Re-call the function but with different arguments
          144|       overrideArgs = mirrorArgs (newArgs: makeOverridable f (overrideWith newArgs));
             |                                           ^
          145|       # Change the result of the function call by applying g to it

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: function 'anonymous lambda' called with unexpected argument 'withWebkit'

       at /home/ilya/nixpkgs/pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/default.nix:1:1:

            1| {
             | ^
            2|   lib,
```

NickCao committed:
{kotatogram-desktop,_64gram,materialgram}: fix overriding withWebkit

NickCao commented:
Fixed, in a ugly way...

ilya-fedin commented:
meh, ofborg seem to not to try to build on darwin (according to https://logs.ofborg.org/?key=nixos/nixpkgs.353524) :(

JohnRTitor reviewed (approved):
```
johnrtitor@darwin01 ~ % nix build "github:nixos/nixpkgs/refs/pull/353524/merge#_64gram" "github:nixos/nixpkgs/refs/pull/353524/merge#kotatogram-desktop" "github:nixos/nixpkgs/refs/pull/353524/merge#materialgram" "github:nixos/nixpkgs/refs/pull/353524/merge#telegram-desktop" --print-out-paths 

/nix/store/b5h15jgzq6hkqwcdajls8ycnb19jh7wl-64gram-1.1.45
/nix/store/24v6afl6i3ay1v70gd1fvyzsh0ss5mnq-kotatogram-desktop-1.4.9-unstable-2024-09-27
/nix/store/hws4kxa355cdwklvriz3mqnc47ala9na-materialgram-5.7.0.1
/nix/store/90rk4m1v7bpgajvnw0mb8s4n1n74xq0s-telegram-desktop-5.7.1

nix build "github:nixos/nixpkgs/refs/pull/353524/merge#_64gram"      5.78s user 15.87s system 0% cpu 1:32:28.51 total
```
